crates/primitives/src/account.rs

```diff
@@ -80,27 +81,29 @@ impl Compact for Bytecode {
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
-        buf.put_u32(self.0.bytecode.len() as u32);
-        buf.put_slice(self.0.bytecode.as_ref());
-        let len = match self.0.state() {
-            BytecodeState::Raw => {
+        let bytecode = &self.0.bytecode()[..];
```

Comment:
This is not correct, bytecode() function would return the first code section.

Hm I see this can be confusing. in Revm we have `original_bytes` and `bytecode` functions, original bytes are needed for analyzed bytecode to remove padding, while `bytecode` represents the first EOF code section. Adding `bytes()` would be better here

Reply:
yeah i didnt account for eof since this would be correct for legacy only